examples/correct/function-overloading.ccp

```diff
@@ -0,0 +1,80 @@
+# basic overloading
+(
+    let f = [] -> Bool => true;
+    let f = [x : Int] -> Int => x;
+    f[];
+    f[1];
+)
+
+# many functions
+(
+    let f = [] -> Bool => true;
+    let f = [x : Int] -> Int => x;
+    let f = [x : Int, y: Int] -> Int => x + y;
+    let f = [x : Int, y: Int, z: Int] -> Int => x + y + z;
+    f[] + f[1] + f[1,1] + f[1,1,1];
+)
+
+# different scopes
+(
+    let f = [] -> Bool => true;
+    (
+        let f = [x : Int] -> Int => x;
+        (
+            let f = [x : Int, y: Int] -> Int => x + y;
+            (
+                let f = [x : Int, y: Int, z: Int] -> Int => x + y + z;
+                f[] + f[1] + f[1,1] + f[1,1,1];
+            )
+        )
+    )
+)
+
+# shadowing/overloading in same scope
+(
+    let f = [] -> Bool => true;
+    let f = [x: Int] => Int => x;
+    let f = [] -> Int => 1;
```

Comment:
Shouldn't this be illegal? There's no way to tell which overload is called with `f[]`. Shadowing works when in nested scope we introduce function/variable with the same name, no?

Reply:
I was under the impression that shadowing works even in the same scope, ie. when declaring new variable with the same name we effectively lose access to previous one.

In this example second f is overload to first one and third f shadows first one, so after 37 we have 2 overloads.

Reply:
sounds a bit cursed, but let's go with that, we'll see what happens